### PR TITLE
V1.4 Developer Guide add `sort` diagrams, update requirements, update use cases for `add`, `edit` and `sort`

### PR DESCRIPTION
**Pull request by UdhayaShan1**
### Third, the use cases for the 'add', 'edit', and 'sort' functionalities have been refined and expanded to ensure clarity and completeness. These updates aim to streamline the development process and improve developer understanding of the project's scope and objectives.

### CODE REVIEW

File: docs/diagrams/BetterModelClassDiagram.puml [hunk 1]
```diff
@@ -4,7 +4,10 @@ skinparam arrowThickness 1.1
 skinparam arrowColor MODEL_COLOR
 skinparam classBackgroundColor MODEL_COLOR
 
-Realodex *-right-> "1" UniquePersonList
+Realodex *--right--> "1" UniqueTagList
+UniqueTagList -right-> "*" Tag
+Person -up-> "1-2 " Tag
```
Comment (appleraincoat) [inline docs/diagrams/BetterModelClassDiagram.puml:9]:
```suggestion
Person -up-> "1..2 " Tag
```

File: docs/diagrams/UndoSequenceDiagram-Logic.puml [hunk 2]
```diff
@@ -24,7 +24,7 @@ activate UndoCommand
 UndoCommand --> AddressBookParser
 deactivate UndoCommand
 
-AddressBookParser --> LogicManager : u
+AddressBookParser --> LogicManager :
 deactivate AddressBookParser
 
 LogicManager -> UndoCommand : execute()
```
Comment (appleraincoat) [inline docs/diagrams/UndoSequenceDiagram-Logic.puml:30]: Can we delete everything related to undo? The undo puml files still show addressbook too

File: docs/diagrams/sort/NewSortSequenceDiagram-Logic.puml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+CommandResult --> DeleteCommand
+deactivate CommandResult
+
+DeleteCommand --> LogicManager : r
+deactivate DeleteCommand
+
+[<--LogicManager
+deactivate LogicManager
+@enduml
```
Comment (appleraincoat) [inline docs/diagrams/sort/NewSortSequenceDiagram-Logic.puml:68]: 👍 
Do remember to delete the old sort sequence diagram.

File: docs/diagrams/sort/SortSequenceDiagram-Model.puml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+participant ":UniquePersonsList" as UniquePersonsList #FF1111
+participant ":ObservableList<Person>" as ObservableList #FF1111
+end box
+
+[-> SortCommand : execute(model)
+activate SortCommand
+
+SortCommand -> Model: getRealodex()
+activate Model
+Model -> SortCommand
```
Comment (appleraincoat) [inline docs/diagrams/sort/SortSequenceDiagram-Model.puml:20]:
```suggestion
Model --> SortCommand
```

Should be dashed line here?

File: docs/diagrams/sort/SortSequenceDiagram-Model.puml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+deactivate Model
+
+SortCommand -> Realodex : getCopyOfInternalListOfUniquePersonsList()
+activate Realodex
+
+Realodex -> UniquePersonsList : getInternalListCopy()
+activate UniquePersonsList
+UniquePersonsList -> Realodex : copied list of persons
```
Comment (appleraincoat) [inline docs/diagrams/sort/SortSequenceDiagram-Model.puml:28]:
```suggestion
UniquePersonsList --> Realodex : copied list of persons
```

similarly also dashed line

File: src/main/java/seedu/realodex/model/person/Birthday.java [hunk 6]
```diff
@@ -109,6 +109,7 @@ public String toString() {
         return optionalBirthday.map(INPUT_DATE_FORMATTER::format).orElse("");
     }
 
+    //@@author UdhayaShan1
```
Comment (appleraincoat) [inline src/main/java/seedu/realodex/model/person/Birthday.java:112]: LOL

**Review by appleraincoat — APPROVED** [review]
LGTM
Do have a look at the dashed lines and remember to delete the old diagrams 😄

File: docs/diagrams/sort/style.puml [hunk 7]
```diff
@@ -0,0 +1,79 @@
+/'
```
Comment (4llysa) [inline docs/diagrams/sort/style.puml:1]: i think it is better to just use the ` styles.puml` in docs/diagrams. no point replicating it in the subfolder

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -479,72 +558,265 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 1u1. Realodex throws an error and highlights the format to user.
+    * 1u2. User enters new data.
+    * Use case resumes from step 1.
+
+* 1v. `Birthday` is blank.
+    * 1v1. Realodex throws an error and highlights the format to user.
+    * 1v2. User enters new data.
+    * Use case resumes from step 1.
+
+* 1w. `Birthday` is blank.
```
Comment (4llysa) [inline docs/DeveloperGuide.md:682]: 1v and 1w are duplicated

File: docs/diagrams/BetterModelClassDiagram.puml (hunk 1 above)
Comment (UdhayaShan1) [reply docs/diagrams/BetterModelClassDiagram.puml:9]: Thanks

File: docs/diagrams/UndoSequenceDiagram-Logic.puml (hunk 2 above)
Comment (UdhayaShan1) [reply docs/diagrams/UndoSequenceDiagram-Logic.puml:30]: Got it

File: docs/diagrams/sort/SortSequenceDiagram-Model.puml (hunk 5 above)
Comment (UdhayaShan1) [reply docs/diagrams/sort/SortSequenceDiagram-Model.puml:28]: Good catch

File: docs/diagrams/sort/style.puml (hunk 7 above)
Comment (UdhayaShan1) [reply docs/diagrams/sort/style.puml:1]: Thanks

File: docs/DeveloperGuide.md (hunk 8 above)
Comment (UdhayaShan1) [reply docs/DeveloperGuide.md:682]: Thanks!

**Pull request merged**